### PR TITLE
Drop buggy TableViewLastColumnResizingFixer class

### PR DESCRIPTION
**Pull request by hebasto**
In Qt 5 the last column resizing with dragging its left edge works out-of-the-box.

The current `TableViewLastColumnResizingFixer` implementation could put the last column content out of the view port and confuse a user:
Screenshot from 2021-01-31 18-04-32

Historical context:
- https://github.com/bitcoin/bitcoin/pull/2862
- https://github.com/bitcoin/bitcoin/pull/3626
- https://github.com/bitcoin/bitcoin/pull/3738
- https://github.com/bitcoin/bitcoin/pull/3920

#205 is a nice addition.

### CODE REVIEW

**Review by jarolrod — COMMENTED** [review]
ACK 3913d1e8c1f604bdd622d5e81e5077ef52b30466, tested on macOS 11.1 Qt 5.15.2

On `master`, when you resize, the last column can go out of the view-port. This can be confusing because there's no contextual clues that there is more to see by scrolling to the right. This `PR` add this contextual awareness by showing a scrollbar. Below are screenshots showing this behavior:

**Master**: Last column can go out of view port while resizing
Screen Shot 2021-02-08 at 3 05 33 AM

**PR**: scrollbar gives contextual awareness that I can scroll right to see more
Screen Shot 2021-02-08 at 3 03 37 AM

**Comment by hebasto** [comment]
@promag @jonatack Do you mind reviewing this PR?

**Review by Talkless — APPROVED** [review]
tACK 3913d1e8c1f604bdd622d5e81e5077ef52b30466, tested on Debian Sid. Can confirm that behavior in previous commit does not produce scroll bar, last column gets "hidden". This PR makes clear that there's more to see in the view.

File: src/qt/receivecoinsdialog.cpp [hunk 1]
```diff
@@ -78,20 +77,19 @@ void ReceiveCoinsDialog::setModel(WalletModel *_model)
...
         tableView->verticalHeader()->hide();
-        tableView->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
         tableView->setModel(_model->getRecentRequestsTableModel());
         tableView->setAlternatingRowColors(true);
         tableView->setSelectionBehavior(QAbstractItemView::SelectRows);
         tableView->setSelectionMode(QAbstractItemView::ContiguousSelection);
         tableView->setColumnWidth(RecentRequestsTableModel::Date, DATE_COLUMN_WIDTH);
         tableView->setColumnWidth(RecentRequestsTableModel::Label, LABEL_COLUMN_WIDTH);
         tableView->setColumnWidth(RecentRequestsTableModel::Amount, AMOUNT_MINIMUM_COLUMN_WIDTH);
+        tableView->horizontalHeader()->setMinimumSectionSize(MINIMUM_COLUMN_WIDTH);
```
Comment (promag) [inline src/qt/receivecoinsdialog.cpp:87]: I've removed this and didn't notice differences, at least on macos.

**Review by promag — COMMENTED** [review]
Tested ACK 3913d1e8c1f604bdd622d5e81e5077ef52b30466 on macos.

File: src/qt/receivecoinsdialog.cpp (hunk 1 above)
Comment (hebasto) [reply src/qt/receivecoinsdialog.cpp:87]: It prevents
Screenshot from 2021-02-21 21-41-20

at least on Linux :)

**Pull request merged**